oci_mlflow/oci_object_storage.py

```diff
@@ -49,43 +110,77 @@ def _download_file(self, remote_file_path, local_path):
         logger.info(f"{full_path}, {remote_file_path}")
         fs.download(full_path, local_path)
 
-    def log_artifact(self, local_file, artifact_path=None):
-        if artifact_path:
-            dest_path = os.path.join(self.artifact_uri, artifact_path)
-        else:
-            dest_path = self.artifact_uri
-        dest_path = os.path.join(dest_path, os.path.basename(local_file))
-        self._upload_file(local_file, dest_path)
+    def log_artifact(self, local_file: str, artifact_path: str = None):
+        """
+        Logs a local file as an artifact, optionally taking an ``artifact_path`` to place it in
+        within the run's artifacts. Run artifacts can be organized into directories, so you can
+        place the artifact in a directory this way.
+
+        Parameters
+        ----------
+        local_file:str
+            Path to artifact to log.
+        artifact_path:str
+            Directory within the run's artifact directory in which to log the artifact.
+        """
+        dest_path = os.path.join(
+            self.artifact_uri, artifact_path or "", os.path.basename(local_file)
+        )
+        ArtifactUploader().upload(local_file, dest_path)
 
-    def log_artifacts(self, local_dir, artifact_path=None):
-        if artifact_path:
-            dest_path = os.path.join(self.artifact_uri, artifact_path)
-        else:
-            dest_path = artifact_path
+    def log_artifacts(self, local_dir: str, artifact_path: str = None):
+        """
+        Logs the files in the specified local directory as artifacts, optionally taking
+        an ``artifact_path`` to place them in within the run's artifacts.
+
+        Parameters
+        ----------
+        local_dir:str
+            Directory of local artifacts to log.
+        artifact_path:str
+            Directory within the run's artifact directory in which to log the artifacts.
+        """
+        artifact_uploader = ArtifactUploader()
+        dest_path = os.path.join(self.artifact_uri, artifact_path or "")
         local_dir = os.path.abspath(local_dir)
-        for (root, _, filenames) in os.walk(local_dir):
+
+        for root, _, filenames in os.walk(local_dir):
             upload_path = dest_path
             if root != local_dir:
                 rel_path = os.path.relpath(root, local_dir)
                 rel_path = relative_path_to_artifact_path(rel_path)
                 upload_path = os.path.join(dest_path, rel_path)
             for f in filenames:
-                self._upload_file(
-                    local_file=os.path.join(root, f),
-                    dest_path=os.path.join(upload_path, f),
+                artifact_uploader.upload(
+                    file_path=os.path.join(root, f),
+                    dst_path=os.path.join(upload_path, f),
                 )
 
     def get_fs(self):
```

Comment:
We can drop this method if it is not used anywhere

Reply:
It is used in `list_artifacts` and `delete_artyifacts` methods.